modules/post/windows/gather/enum_ad_users.rb

```diff
@@ -0,0 +1,174 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'rex'
+require 'msf/core'
+
+class Metasploit3 < Msf::Post
+  include Msf::Auxiliary::Report
+  include Msf::Post::Windows::LDAP
+  include Msf::Post::Windows::Accounts
+
+  UAC_DISABLED = 0x02
+
+  def initialize(info = {})
+    super(update_info(
+      info,
+      'Name'         => 'Windows Gather Active Directory Users',
+      'Description'  => %{
+        This module will enumerate user accounts in the default Active Domain (AD) directory and stores
+      them in the database.
+      },
+      'License'      => MSF_LICENSE,
+      'Author'       => [
+        'Ben Campbell',
+        'Carlos Perez <carlos_perez[at]darkoperator.com>'
+      ],
+      'Platform'     => [ 'win' ],
+      'SessionTypes' => [ 'meterpreter' ]
+    ))
+
+    register_options([
+      OptBool.new('STORE_LOOT', [true, 'Store file in loot.', false]),
+      OptBool.new('EXCLUDE_LOCKED', [true, 'Exclude in search locked accounts..', false]),
+      OptBool.new('EXCLUDE_DISABLED', [true, 'Exclude from search disabled accounts.', false]),
+      OptEnum.new('UAC', [true, 'Filter on User Account Control Setting.', 'ANY',
+                          [
+                            'ANY',
+                            'NO_PASSWORD',
+                            'CHANGE_PASSWORD',
+                            'NEVER_EXPIRES',
+                            'SMARTCARD_REQUIRED',
+                            'NEVER_LOGGEDON'
+                          ]])
+    ], self.class)
+  end
+
+  def run
```

Comment:
this method can be broken up into smaller atomic methods, building the filter can be one method, doing the query another method, parsing the results a third. should be fairly easy to refactor this